cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/AppFabricTestBase.java

```diff
@@ -506,18 +506,42 @@ protected HttpResponse addAppArtifact(Id.Artifact artifactId, Class<?> cls) thro
   protected HttpResponse addPluginArtifact(Id.Artifact artifactId, Class<?> cls,
                                            Manifest manifest,
                                            Set<ArtifactRange> parents) throws Exception {
+    return addPluginArtifact(artifactId, cls, manifest, parents, null);
+  }
 
+  /**
+   * adds a plugin artifact. This method is present for testing invalid plugin class json and it usage is not
+   * recommended.
+   * @param artifactId the artifact id
+   * @param cls the application class
+   * @param manifest manifest
+   * @param parents parents range of artifact
+   * @param pluginClassesJson JSON array represetation of a collection of plugin class
+   * @return {@link HttpResponse} recieved response
+   * @throws Exception
+   */
+  protected HttpResponse addPluginArtifact(Id.Artifact artifactId, Class<?> cls,
+                                           Manifest manifest,
+                                           @Nullable Set<ArtifactRange> parents,
+                                           @Nullable  String pluginClassesJson) throws Exception {
```

Comment:
extra space

Reply:
is there a reason this is a string and not a `Set<PluginClass>`?

Reply:
Yes, want to test to with invalid JSON. Don't want to make PluginClass default constructor public to be able to construct an invalid object here.